File: frontend/src/lib/components/definitions/qt-resource-rebuilder.ts
import { ComponentDefinition } from '../types'
import { CommandBuilders } from '../utils'
import { CommonHooks } from '../hooks'

export const qtResourceRebuilderComponent: ComponentDefinition = {
  id: 'qt-resource-rebuilder',
  name: 'Qt Resource Rebuilder',
  description: 'Enables UI modifications like rM Hacks',
  version: 'v16-14112025',

  dependencies: ['xovi'],

  checkInstalled: () =>
    CommandBuilders.testExists('/home/root/xovi/extensions.d/qt-resource-rebuilder.so'),

  install: () =>
    CommandBuilders.copyFile(
      '/home/root/extensions/qt-resource-rebuilder.so',
      '/home/root/xovi/extensions.d/qt-resource-rebuilder.so'
    ),

  postInstall: CommonHooks.rebuildHashtableDialog(),

  uninstall: () =>
    CommandBuilders.removeFile('/home/root/xovi/extensions.d/qt-resource-rebuilder.so'),

  category: 'core',
  tags: ['framework', 'ui-modification'],
  requiresUserConfirmation: true,
}

File: internal/component/definitions/qt_resource_rebuilder.go
package definitions

import (
	"reManager/internal/commands"
	"reManager/internal/component"
	"reManager/internal/installer"
)

var QtResourceRebuilderComponent = &component.ComponentDefinition{
	ID:          "qt-resource-rebuilder",
	Name:        "Qt Resource Rebuilder",
	Description: "Enables UI modifications like rM Hacks",
	Version:     "v16-14112025",

	Dependencies: []string{"xovi"},

	CheckInstalled: func(ctx component.CommandContext) component.CommandResult {
		return commands.TestExists("/home/root/xovi/extensions.d/qt-resource-rebuilder.so", commands.PathFile)
	},

	Install: func(ctx component.CommandContext) []component.CommandResult {
		return []component.CommandResult{
			commands.CopyFile(
				"/home/root/extensions/qt-resource-rebuilder.so",
				"/home/root/xovi/extensions.d/qt-resource-rebuilder.so",
			),
		}
	},

	PostInstall: installer.RebuildHashtableDialog(),

	Uninstall: func(ctx component.CommandContext) []component.CommandResult {
		return []component.CommandResult{
			commands.RemoveFile("/home/root/xovi/extensions.d/qt-resource-rebuilder.so"),
		}
	},

	Category:                 "core",
	Tags:                     []string{"framework", "ui-modification"},
	RequiresUserConfirmation: true,
}
